src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -16,32 +16,21 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal partial class Binder
     {
-        /// <summary>
-        /// Only handles assignment-only or declaration-only deconstructions at this point.
-        /// Issue https://github.com/dotnet/roslyn/issues/15050 tracks allowing mixed deconstructions
-        /// </summary>
         private BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
         {
             var left = node.Left;
             var right = node.Right;
+            CSharpSyntaxNode declaration = null;
+            CSharpSyntaxNode expression = null;
+            var result = BindDeconstruction(node, left, right, diagnostics, ref declaration, ref expression);
+            if (declaration != null && expression != null)
+            {
+                // We only allow assignment-only or declaration-only deconstructions at this point.
+                // Issue https://github.com/dotnet/roslyn/issues/15050 tracks allowing mixed deconstructions.
+                // For now we give an error when you mix.
+                Error(diagnostics, ErrorCode.ERR_MixedDeconstructionUnsupported, left);
```

Comment:
There already is one.

---
In reply to: [89845635](https://github.com/dotnet/roslyn/pull/15548#discussion_r89845635) [](ancestors = 89845635)